doc/architecture.md

```diff
@@ -1,109 +1,176 @@
+- CSV - data from a .csv formatted file
+- Redis - data from Redis server channels
+- SBN - data via the Software Bus Network attached to an instance of [NASA's core Flight Software](https://github.com/nasa/cFS "cFS on GitHub")
 
-- **[DATA_HANDLING]** - The telemetry data parser, it is a required section and the following key is required
-  - DataSourceFile (string) - full path and python file name of the [OnAir data source](#the-onairdatasource) to use
+However, all of these just implement the abstract class [`OnAirDataSource`](https://github.com/nasa/OnAIR/blob/686df367bf4b9679ee9be11524230e99a499e5f0/onair/data_handling/on_air_data_source.py#L4 "on_air_data_source.py") and users are able (nay, encouraged!) to write their own.
 
-- **[PLUGINS]** - The various [plugins](#the-plugins) that OnAIR will use to construct the [pipeline](#the-pipeline), it is a required section and all following keys are required
-  - KnowledgeRepPluginDict (dict)
-  - LearnersPluginDict (dict)
-  - PlannersPluginDict (dict)
-  - ComplexPluginDict (dict)
+### [CSV Parser](https://github.com/nasa/OnAIR/blob/686df367bf4b9679ee9be11524230e99a499e5f0/onair/data_handling/csv_parser.py#L4 "csv_parser.py")
+When you have telemetry data in CSV (Comma-Separated Values) format that you want to use with the OnAIR platform.
 
-  Each plugin dictionary consists of key, name for the plugin at runtime, to value, path to the plugin's base directory. The key name is not required to be the same as the name of the plugin, but the directory used for the plugin must contain the concrete plugin class named **\<directory name\>_plugin.py**. Multiple of the same plugin directory may be used by defining unique key names, e.g. {"Reporter1":"plugins/reporter", "Reporter2":"plugins/reporter"} will provide two reporter plugins. A plugin type may be left empty by giving it an empty dict, **{}**.
+#### CSV File Requirements
+What a file must have to be interpreted correctly.
 
-- **[OPTIONS]** - Provides for any optional settings, section and its key are optional
-  - IO_Enabled (bool) - this determines whether or not to provide text output during runtime, omission of this equates to **_false_**
+- The first row shall contain headers that match the order from the [telemetry file](#telemetry-json-file)
+- Each subsequent row shall contain the values that represent a single frame of data
 
-### Telemetry Definitions (.json) File
-Also known as the 'Metafile', this provides the information necessary for OnAIR to understand the received data.
+#### CSV File Considerations
+What should be known about file parsing and usage.
 
-## The OnAirDataSource
-This is the adapter that attaches OnAIR to a telemetry source. It provides the basic data that is used by the cognitive components to render reasoning about the system.
+- Acceptable values, all will become floats
+  - Integer
+  - Float
+  - Timestamp str
```

Comment:
Maybe just refer to convert_str_to_timestamp in parser_util.py to show the expected formats

Reply:
Done.